### ISSUE TITLE
The server does not start after update to fastify v4.28.1

### ISSUE THREAD

**Issue by pasha-vuiko** [geut/fastify-uws#6]
The server does not start after update to fastify v4.28.1, previously worked fine. The server just being freezed on call of .listen()

**Comment by tinchoz49** [comment]
Hey @pasha-vuiko I'm going to check what's going on

**Comment by pasha-vuiko** [comment]
By the way, I use MacOS 14.5 and Node.js v20.15.0

**Comment by dpaez** [comment]
Hey @pasha-vuiko could you add more info or share some code. 

I just tested this with node v20.15.1 and the example code and is working as expected. One can access the GET endpoint and open local websockets and send messages normally.

**EDIT**: just to clarify this issue and my comment. The server starts OK and the endpoints/uws works as expected. It seems like something changed between fastify `v4.27.x` and `v4.28.x`, because the `onListen` hook does not trigger after the server starts in latest fastify.

**Comment by pasha-vuiko** [comment]
Hi @dpaez!

Here is the sample code:

```javascript
import fastify from "fastify";
import { serverFactory } from "@geut/fastify-uws";

const app = fastify({
  serverFactory,
});

app.get("/", async (request, reply) => {
  const hello = 'hello';

  return { hello: hello };
});

app
  .listen({
    host: "0.0.0.0",
    port: 3000,
  })
  .then((address) => {
    console.log(`server listening on ${address}`);
  });
```

Yeah, indeed looks like server starts, but .listen() promise is not being fulfilled, so I don't see the console.log() in a terminal. Looks like it caused by the changes in Fastify:
https://github.com/fastify/fastify/pull/5523/files

**Comment by tinchoz49** [comment]
@dpaez that's right guys, fastify change the way of how it handles the listening callback, i'm working on a fix because other things change too.

**Comment by tinchoz49** [comment]
fixed in v5.0.0
